tortoisebeacon/handlers.go

```diff
@@ -183,7 +183,7 @@ func (tb *TortoiseBeacon) verifyProposalMessage(m ProposalMessage, currentEpoch
 		return fmt.Errorf("calculate proposal: %w", err)
 	}
 
-	if !tb.vrfVerifier(m.MinerID.VRFPublicKey, currentEpochProposal, m.VRFSignature) {
+	if !tb.vrfVerifier.Verify(signing.NewPublicKey(m.MinerID.VRFPublicKey), currentEpochProposal, m.VRFSignature) {
```

Comment:
@nkryuchkov this part is not finished, right? we need to recover public key from signature here and verify that miner is in the active set

Reply:
It works with the `MinerID` field too, but the field is redundant, we can recover it from the signature. There's a check that miner has an ATX in a previous epoch [here](https://github.com/dshulyak/go-spacemesh/blob/finalize-weakcoin/tortoisebeacon/handlers.go#L83).

Reply:
great, i will leave it as is then for this PR. i thought that needs to be done because you mentioned it previously